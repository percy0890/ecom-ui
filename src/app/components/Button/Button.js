import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

export default function Button({
  buttonType,
  customClass,
  customInvertClass,
  icon,
  label,
  reverse,
  onClick,
  invert,
  isDisabled,
}) {
  const reverseButtonClass = reverse ? 'btn-reverse' : '';
  const buttonTypeClass = buttonType
    ? `btn-${buttonType.trim().toLowerCase()}`
    : '';

  const labelLessClass = label ? '' : 'label-less';
  const invertClass = invert ? 'invert' : '';
  return (
    <button
      type="button"
      className={`btn ${buttonTypeClass} ${customClass} ${invertClass} ${invert &&
        customInvertClass} ${labelLessClass}`}
      onClick={onClick}
      disabled={isDisabled}
    >
      <div className={` btn-content-container ${reverseButtonClass}`}>
        {label && <span className="label">{label}</span>}
        {icon && (
          <div className="icon-container">
            <img src={icon} alt="button icon" className="icon" />
          </div>
        )}
      </div>
    </button>
  );
}

Button.propTypes = {
  buttonType: PropTypes.string.isRequired,
  icon: PropTypes.string,
  label: PropTypes.string,
  customClass: PropTypes.string,
  customInvertClass: PropTypes.string,
  reverse: PropTypes.bool,
  invert: PropTypes.bool,
  onClick: PropTypes.func,
  isDisabled: PropTypes.bool,
};
