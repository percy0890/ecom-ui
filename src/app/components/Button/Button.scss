@import '~app/styles/core.scss';

.btn {
  border: 1px solid transparent;
  color: $app-text-white;
  font-size: 1.4rem;
  cursor: pointer;
  padding: 1.3rem 1.5rem;
  font-family: $app-font-bold;
  border-radius: 3px;

  &:hover {
    opacity: 0.9;
  }

  .btn-content-container {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  &.label-less {
    padding: 1rem;
    outline: none;
  }

  &-primary {
    background-color: $brand-color-primary;
    color: $app-text-white;
    box-shadow: $app-shadow-element;
    border: 1px solid $brand-color-primary;

    &.invert {
      color: $brand-color-primary;
      background-color: $app-bg-white;
    }
  }

  &-secondary {
    background-color: $brand-color-secondary;
    color: $app-text-white;
    box-shadow: $app-shadow-element;
    border: 1px solid $brand-color-secondary;

    &.invert {
      color: $brand-color-secondary;
      background-color: $app-bg-white;
    }
  }

  &-transparent {
    background-color: transparent;
    color: $brand-color-primary;
  }

  &-default {
    background-color: $app-text-white;
    color: $app-text-primary;
    border: 1px solid $app-border-color;
    box-shadow: $app-shadow-element;

    &.invert {
      color: $app-bg-white;
      background-color: $app-text-primary;
    }
  }

  &-inactive {
    background-color: $app-bg-container;
    color: $app-text-primary;
    box-shadow: $app-shadow-element;
  }

  .icon-container {
    display: flex;
    margin-left: 0.5rem;

    .icon {
      height: 1.6rem;
    }
  }

  .label {
    font-size: 1.8rem;
    font-family: $app-font-bold;
  }

  .label-less .icon-container {
    margin: 0;
  }

  &-reverse {
    flex-direction: row-reverse;

    .icon-container {
      margin-left: 0;
      margin-right: 0.5rem;
    }
  }
}
