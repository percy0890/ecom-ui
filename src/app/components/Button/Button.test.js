import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    expect(
      shallow(<Button buttonType="" reverse invert customInvertClass="btn" />),
    ).toMatchSnapshot();
  });
});
