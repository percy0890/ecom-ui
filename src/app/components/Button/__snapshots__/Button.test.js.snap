// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="btn  undefined invert btn label-less"
  type="button"
>
  <div
    className=" btn-content-container btn-reverse"
  />
</button>
`;
